xline/src/storage/kvstore.rs

```diff
@@ -160,55 +225,111 @@ impl KvStoreInner {
         );
     }
 
+    /// Get a `KeyRevision` if the key is not deleted
+    fn get_index(indexes: &[KeyRevision]) -> Option<KeyRevision> {
+        indexes.last().filter(|index| !index.is_deleted()).copied()
+    }
+
     #[allow(clippy::pattern_type_mismatch)]
     /// Handle `RangeRequest`
-    fn handle_range_request(
-        btree: &MutexGuard<BTreeMap<Vec<u8>, KeyValue>>,
-        req: RangeRequest,
-    ) -> RangeResponse {
+    fn handle_range_request(&self, req: RangeRequest) -> RangeResponse {
+        let key_index = self.index.lock();
+        let db = self.db.lock();
```

Comment:
If they're always locked at the same time, shall we move them into one lock?

Reply:
For several locks acquired at the same time, we'd better document the order somewhere to make sure we keep the order everywhere.

Reply:
I will separate index and db in to different module in next patch. Lock will be acquired in their own module.